graylog2-web-interface/src/components/streamrules/StreamRulesEditor.jsx

```diff
@@ -79,83 +107,87 @@ class StreamRulesEditor extends React.Component {
+      <>
+        <MatchIcon empty name="exclamation-circle" /> Please load a message in Step 1 above to check if it would match against these rules.
+      </>
+    );
   };
 
   render() {
-    const styles = (this.state.matchData ? this._getListClassName(this.state.matchData) : 'info');
-    if (this.state.stream && this.state.streamRuleTypes) {
+    const { matchData, stream, streamRuleTypes } = this.state;
+    const { currentUser, messageId, index } = this.props;
+    const styles = (matchData ? this._getListClassName(matchData) : 'info');
+
+    if (stream && streamRuleTypes) {
       return (
-        <div className="row content">
-          <div className="col-md-12 streamrule-sample-message">
-            <h2>
-              1. Load a message to test rules
-            </h2>
+        <Row className="content">
+          <Col md={12} className="streamrule-sample-message">
+            <h2>1. Load a message to test rules</h2>
 
             <div className="stream-loader">
-              <LoaderTabs messageId={this.props.messageId} index={this.props.index} onMessageLoaded={this.onMessageLoaded} />
-            </div>
-
-            <div className="spinner" style={{ display: 'none' }}><h2><Icon name="spinner" spin /> &nbsp;Loading message
-            </h2>
-            </div>
-
-            <div className="sample-message-display" style={{ display: 'none', marginTop: '5px' }}>
-              <strong>Next step:</strong>
-              Add/delete/modify stream rules in step 2 and see if the example message would have been
-              routed into the stream or not. Use the button on the right to add a stream rule.
+              <LoaderTabs messageId={messageId}
+                          index={index}
+                          onMessageLoaded={this.onMessageLoaded} />
             </div>
 
             <hr />
 
             <div className="buttons pull-right">
-              <button className="btn btn-success show-stream-rule" onClick={this._onAddStreamRule}>
+              <Button bsStyle="success"
+                      className="show-stream-rule"
+                      onClick={this._onAddStreamRule}>
                 Add stream rule
-              </button>
+              </Button>
               <StreamRuleForm ref={(newStreamRuleForm) => { this.newStreamRuleForm = newStreamRuleForm; }}
                               title="New Stream Rule"
-                              streamRuleTypes={this.state.streamRuleTypes}
+                              streamRuleTypes={streamRuleTypes}
                               onSubmit={this._onStreamRuleFormSubmit} />
             </div>
 
-            <h2>
-              2. Manage stream rules
-            </h2>
+            <h2>2. Manage stream rules</h2>
 
-            {this._explainMatchResult()}
+            <MatchingTypeSwitcher stream={stream} onChange={this.loadData} />
+            <Panel bsStyle={styles}>
+              <StreamAlertHeader>{this._explainMatchResult()}</StreamAlertHeader>
```

Comment:
It looks like when trying to load the "Manage rules" page there's an error:

```
Objects are not valid as a React child (found: object with keys {$$typeof, render, displayName, attrs, componentStyle, foldedComponentIds, styledComponentId, target, withComponent, warnTooManyClasses, toString}). If you meant to render a collection of children, use an array instead.
    in templateFunction (created by StreamRulesEditor)
    in h4 (created by Context.Consumer)
    in StyledComponent (created by StreamRulesEditor__StreamAlertHeader)
    in StreamRulesEditor__StreamAlertHeader (created by StreamRulesEditor)
    in div (created by Panel)
    in div (created by Panel)
    in Panel (created by Context.Consumer)
    in StyledComponent (created by Panel__StyledPanel)
    in Panel__StyledPanel (created by StreamRulesEditor)
    in div (created by Col)
    in Col (created by StreamRulesEditor)
    in div (created by Row)
    in Row (created by StreamRulesEditor)
    in StreamRulesEditor (created by StreamEditPage)
```

Reply:
Bad parenthesis wrap in a styled-component 😩